---
output:
  bookdown::pdf_document2:
    includes:
      before_body: ../template/doc_prefix.tex
      in_header: ../template/preamble.tex
    keep_tex: yes
    latex_engine: pdflatex
    number_sections: no
    toc: no
  bookdown::html_document2:
    number_sections: no
    theme: readable
    toc: yes
  bookdown::tufte_html2:
    number_sections: no
    toc: yes
  bookdown::word_document2: null
fontsize: 10pt
linestretch: 1.5
link-citations: yes
bst: ../template/newphyt.bst
bibliography: stomata-independence.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
knitr::opts_knit$set(root.dir = "..")
```

```{r}
library(dplyr)
library(glue)
library(lubridate)
library(readr)
library(ropenstomata)
library(stringr)
library(tibble)

source("r/functions.R")

# Initial size of data set
stomata_anatomy1 = stomata_anatomy |> 
  filter(!str_detect(trait, "width"), 
         !(source == "pathare_increased_2020" & str_detect(trait, "pore")))

processed_data_files = tibble(file_name = list.files("processed-data"))
objects_files = tibble(file_name = list.files("objects"))

# 01_taxize-data ----
taxize_output = read_rds("processed-data/taxize_output.rds") |>
  mutate(ref = case_when(
    data_source_title == "GRIN Taxonomy for Plants" ~ "united_states_department_of_agriculture_agricultural_research_service_germplasm_2020",
    data_source_title == "NCBI" ~ "schoch_ncbi_2020",
    data_source_title == "Open Tree of Life Reference Taxonomy" ~ "rees_automated_2017",
    data_source_title == "The International Plant Names Index" ~ "the_royal_botanic_gardens_international_2020",
    data_source_title == "Tropicos - Missouri Botanical Garden" ~ "missouri_botanical_garden_tropicos_2020",
  ))

taxize_source_ref = taxize_output |> 
  mutate(x = glue("{data_source_title} [@{ref}]")) |>
  pull(x) |>
  unique() |>
  sort() |>
  str_c(collapse = ", ")

n_tax1 = length(unique(taxize_output$user_supplied_name))

# 02_resolve-names ----
resolved_names = read_rds("processed-data/resolved_names.rds")

n_tax2 = length(unique(resolved_names$resolved_name))

# 03_make-megatree ----
phy = read_rds("processed-data/phy.rds")

# 04_prepare-subtrees ----
polytomies = read_rds("objects/polytomies.rds")

# 06_connect-data ----
trimmed_data = read_rds("processed-data/trimmed-data.rds")
trimmed_phy = read_rds("processed-data/trimmed-phylogeny.rds")

n_tax3 = nrow(trimmed_data)

# 07_make-pairs ----
pair_div = read_rds("objects/pair_div.rds")
n_pair = nrow(pair_div)

# 10_fit-pairs ----
fit = read_rds("objects/fit.rds")

# 14_plot-h1 ----
h1_parameters = read_rds("objects/h1_parameters.rds") |>
  mutate(
    estimate = scientize(value),
    lower = scientize(.lower),
    upper = scientize(.upper),
    interval = glue("95\\% HPD interval $[{lower},{upper}]$")
  )

# 15_plot-h2 ----
h2_parameters = read_rds("objects/h2_parameters.rds") |>
  mutate(
    estimate = scientize(value, threshold = -2L),
    lower = scientize(.lower, threshold = -2L),
    upper = scientize(.upper, threshold = -2L),
    interval = glue("95\\% HPD interval $[{lower},{upper}]$")
  )

```

\begin{flushleft}
{\Large
\textbf\newline{Developmental integration cannot explain major features of stomatal anatomical evolution in seed plants}
}
\newline
\\
Christopher D. Muir\textsuperscript{1,*},
Miquel \`{A}ngel Conesa\textsuperscript{2},
Jeroni Galm\'{e}s\textsuperscript{2},
Varsha S. Pathare\textsuperscript{3},
Patricia Rivera\textsuperscript{4},
Rosana López Rodríguez\textsuperscript{5},
Teresa Terrazas\textsuperscript{4},
Dongliang Xiong\textsuperscript{6}
\\
\bigskip
\bf{1} School of Life Sciences, University of Hawaiʻi at M\=anoa, Honolulu, HI 96822, USA \\
\bf{2} Research Group on Plant Biology under Mediterranean Conditions, Departament de Biologia, Universitat de les Illes Balears, Ctra. Valldemossa km 7.5, E-07122, Palma, Spain \\
\bf{3} School of Biological Sciences, Washington State University, Pullman, WA 99164-4236, USA \\
\bf{4} Departamento de Botánica, Instituto de Biología, Universidad Nacional Autónoma de México, Apartado Postal 70‑367, 04510 Mexico City, Mexico \\
\bf{5} Departamento de Sistemas y Recursos Naurales, Universidad Politécnica de Madrid, 28040 Madrid, Spain. \\
\bf{6} National Key Laboratory of Crop Genetic Improvement, MOA Key Laboratory of Crop Ecophysiology and Farming System in the Middle Reaches of the Yangtze River, College of Plant Science and Technology, Huazhong Agricultural University, Wuhan, Hubei 430070, China\\
\bigskip
* cdmuir@hawaii.edu

\end{flushleft}

# Abstract

* Developmental integration can cause traits to covary over macroevolutionary time and in some cases prevent populations from reaching their adaptive optima. Developmental integration between stomatal size and density may contribute to two major features of stomatal anatomical evolution: inverse size-density scaling and bimodal stomatal ratio. If these patterns result from developmental integration, we predicted that in amphistomatous leaves 1) stomatal size and density should covary similarly on both abaxial and adaxial surfaces and 2) stomatal traits (size and density) on each surface should covary isometrically.
* We synthesized data on stomatal density and length from amphistomatous leaves of `r n_tax3` terrestrial seed plant taxa mostly from the literature. We estimated the covariance in divergence between stomatal traits from `r n_pair` phylogenetically independent contrasts using a robust Bayesian model.
*	Adaxial stomatal density, but not length, is evolutionary labile and not strongly integrated with stomatal length or abaxial stomatal density. Hence, developmental integration alone cannot explain inverse size-density scaling nor bimodal stomatal  ratio.
* Quasi-independent evolution of stomatal anatomical traits facilitates largely unfettered access to fitness optima. If stomatal anatomical traits are near their current fitness optimum, this implies that limits on trait (co)variance result from selective rather than developmental constraints. However, we cannot rule out that developmental integration is important in some lineages. Future research should identify the mechanistic basis of (dis)integration in stomatal development.

**Keywords:** Adaptation, amphistomy, developmental integration, leaf, phylogenetic comparative methods, stomata

\linenumbers

# Introduction

The ability for traits to evolve independently of one another is a necessary prerequisite for adaptation to complex environments [@lewontin_adaptation_1978]. If traits can evolve independently and there is sufficient genetic variation, then selection should move populations toward their multivariate phenotypic optimum. Adaptive evolution may be constrained if traits cannot evolve independently because they are developmentally integrated. Developmentally integrated traits have a "disposition for covariation" [@armbruster_integrated_2014], meaning that evolutionary divergence between lineages in one character will be tightly associated with divergence in another character. Allometry is a classic example of developmental integration that may constrain phenotypic evolution (reviewed in @pelabon_evolution_2014). Strong allometric covariation between traits within populations can constrain macroevolutionary divergence for long periods of time depending on the strength and direction of selection [@lande_quantitative_1979]. However, developmental integration does not necessarily hamper adaptation, and can even accelerate adaptive evolution when trait covariation is aligned with the direction of selection [@hansen_is_2003]. For example, fusion of floral parts increases their developmental integration which may increase the rate and precision of multivariate adaptation to specialist pollinators (Berg’s rule) [@berg_general_1959; @berg_ecological_1960; @conner_raissa_2014; @armbruster_covariance_1999]. In this study we are interested in quantifying the strength of developmental integration in stomatal anatomy and whether developmental integration might hamper or accelerate adaptive evolution.

Stomata are microscopic pores on the leaf or other photosynthetic surfaces of most land plants formed by a pair of guard cells. Here we limit our focus to stomatal traits on leaves within terrestrial seed plants, primarily angiosperms. The density, size, and patterning of stomata on a leaf set the maximum stomatal conductance to CO$_2$ diffusing into a leaf and the amount of water that transpires from it [@sack_hydrology_2003; @franks_effect_2001; @galmes_leaf_2013]. Plants typically operate below their anatomical maximum by dynamically regulating stomatal aperture. Even though operational stomatal conductance determines the realized photosynthetic rate and water-use efficiency, anatomical parameters are useful in that they set the range of stomatal function [@de_boer_optimal_2016] and are correlated with actual stomatal function under natural conditions [@murray_consistent_2020].

Two salient features of stomatal anatomy have been recognized for decades but we do not yet understand the evolutionary forces that generate and maintain them. We denote these two features as “inverse size-density scaling” and “bimodal stomatal ratio”  (Fig. \ref{fig:concepts}). Inverse size-density scaling refers to the negative interspecific correlation between the size of the stomatal apparatus and the density of stomata [@weiss_untersuchungen_1865; @franks_maximum_2009; @de_boer_optimal_2016; @sack_developmental_2016; @liu_scaling_2021]. Across species, leaves with smaller stomata tend to pack them more densely, but there is significant variation about this general trend. Bimodal stomatal ratio refers to the observation that the ratio of stomatal density on the adaxial (upper) surface to the density on the abaxial (lower) has distinct modes (Fig. 1). Stomata are most often found only on the lower leaf surface (hypostomy), but occur on both surfaces (amphistomy) in some species [@metcalfe_anatomy_1950; @parkhurst_adaptive_1978; @mott_adaptive_1982], especially herbaceous plants from open, high light habitats [@salisbury_i_1928; @mott_adaptive_1982; @gibson_structure-function_1996; @smith_associations_1998; @jordan_using_2014; @muir_making_2015; @bucher_stomatal_2017; @muir_light_2018]. @muir_making_2015 described bimodal stomatal ratio formally but the pattern is apparent in earlier comparative studies of the British flora (*cf* [@peat_comparative_1994] Fig 1). For both features, we limit our focus in this study to interspecific variation in mean trait values and do not seek to understand intraspecifc variation.

\begin{figure}[ht]
\includegraphics[width=\textwidth]{../figures/concepts.pdf}
\caption{Two salient features of stomatal anatomy in land plants are the (a) inverse relationship between stomatal size and density and (b) the bimodal distribution of stomatal ratio. At broad phylogenetic scales, leaves with smaller stomata ($x$-axis, log-scale) tend to have greater stomatal density  ($x$-axis, log-scale), but there is a lot of variation about the overall trend indicated by the grey ellipse. Hypostomatous leaves (stomatal ratio = 0) are more common than amphistomatoues leaves, but within amphistomatous leaves, the density of stomata on each surface tends to be similar (stomatal ratio $\approx$ 0.5), which we refer to as bimodal stomatal ratio.}
\label{fig:concepts}
\end{figure}

Given the significance of stomata for plant function and global vegetation modeling, we would like to understand whether these major anatomical features are shaped primarily by adaptive or nonadaptive evolutionary forces. We briefly review adaptive hypotheses for both inverse size-density scaling and bimodal stomatal ratio. Then we discuss how developmental integration might contribute to these features, in addition to or in lieu of adaptive evolution. Stomatal size and density determine the maximum stomatal conductance to CO$_2$ and water vapor but also take up space on the epidermis, which could be costly for both construction and maintenance. Natural selection should favor leaves that have enough stomata of sufficient size to supply CO$_2$ for photosynthesis. Hence leaves with few, small stomata and high photosynthetic rates do not exist because they would not supply enough CO$_2$. Conversely, excess stomata or extra large stomata beyond the optimum may result in stomatal interference [@zeiger_stomatal_1987], incur metabolic costs [@deans_optimization_2020], and/or risk hydraulic failure [@henry_stomatal_2019]. The distribution of stomata and density may therefore represent the combinations that ensure enough, but not too much, stomatal conductance. @franks_maximum_2009 further hypothesized that the evolution of small stomata in angiosperms enabled increased stomatal conductance while minimizing the epidermal area allocated to stomata. 

Developmental integration between stomatal size and density, mediated by meristematic cell volume, could also explain inverse size-density scaling. If the size of both guard cells and the size of pavement cells between stomata are determined by initial meristematic cell volume, then changes in cell volume early in leaf development would cause both increased stomatal size and lower density [@brodribb_unified_2013]. This type of developmental integration would not hinder adaptive evolution when the main axes of selection were aligned with the developmental correlation. For example, if higher maximum stomatal conductance were achieved primarily by increasing stomatal density and decreasing stomatal size as proposed by @franks_maximum_2009, then developmental integration might accelerate the response to selection compared to a case where stomatal size and density are completely independent.

The adaptive significance of variation in stomatal ratio is unknown, but we have some clues based on the distribution of hypo- and amphistomatous leaves. The functional significance of amphistomy is that it adds a second parallel pathway from the substomatal cavities through the leaf internal airspace to sites of carboxylation in the mesophyll [@parkhurst_adaptive_1978; @gutschick_photosynthesis_1984]. Thus amphistomatous leaves have lower resistance to diffusion through the airspace which increases the photosynthetic rate [@parkhurst_intercellular_1990]. Despite this amphistomy advantage, most leaves are hypostomatous, suggesting that the benefits of amphistomy in terms of increased photosynthesis usually do not outweigh the costs of stomata on the upper surface. Amphistomy should increase photosynthesis under saturating-light conditions where CO$_2$ supply limits photosynthesis. This may explain why amphistomatous leaves are most common in high light habitats [@salisbury_i_1928; @mott_adaptive_1982; @gibson_structure-function_1996; @smith_associations_1998; @jordan_using_2014; @muir_making_2015; @bucher_stomatal_2017], especially in herbs [@muir_light_2018]. However, the light environment alone cannot explain why hypostomatous leaves predominate in shade plants [@muir_is_2019], suggesting that we need to understand the costs of upper stomata better. Upper stomata increase the susceptibility to rust pathogens in *Populus* [@mckown_association_2014; @mckown_role_2019; @fetter_growthdefense_2021]. Amphistomy may also cause the palisade mesophyll to dry out under strong vapor pressure deficits [@buckley_how_2015]. Other hypotheses about the adaptive significance of stomatal ratio are discussed in @muir_making_2015 and @drake_two_2019.

A striking feature of the interspecific variation in stomatal ratio is that trait values are not uniformly distributed, but strongly bimodal. @muir_making_2015 derived general conditions in which bimodality arises because adaptive optima are restricted to separate regimes, but this model has not been tested. An alternative hypothesis is that stomatal traits on the ab- and adaxial surfaces are developmentally integrated because stomatal development is regulated the same way on each surface. In hypostomatous leaves, stomatal development is turned off in the adaxial surface. In amphistomatous leaves, stomatal development proceeds on both surfaces, but evolutionary changes in stomatal development affect traits on both surfaces because they are tethered by a shared developmental program. This model of developmental integration would lead to a bimodal trait distribution because leaves would either be hypostomatous (stomatal ratio equal to 0) or have similar densities on each surface (stomatal ratio approximately 0.5). To our knowledge, this hypothesis has not been put forward in the literature but came about during discussion with one of us (Muir, personal communication with EJ Edwards).

We reasoned that the developmental hypotheses for inverse size-density scaling and bimodal stomatal ratio could be tested using amphistomatous leaves. Since both leaf surfaces are formed from the same genome and meristematic cells, if developmental integration is strong we would expect similar patterns of trait covariation in the abaxial and adaxial surface (see below for specific predictions). Conversely, if traits covary differently on each surface it would indicate that stomatal anatomical traits can be developmentally disintegrated. Analogously, variation in the genetic correlation and interspecific divergence of sexually dimorphic traits in dioecious species demonstrate that integration is not fixed and can be modified by selection [@barrett_sexual_2013]. Below we reiterate the hypotheses and specific predictions for amphistomatous leaves:

\noindent {\bf{Inverse size-density scaling}}

- *Hypothesis*: Meristematic cell volume mediates developmental integration between stomatal size and density 

- *Prediction*: Amphistomatous leaves will exhibit identical size-density scaling on each surface

\noindent {\bf{Bimodal stomatal ratio}}

- *Hypothesis*: Stomatal traits on both leaf surfaces are developmentally integrated because they follow the same developmental program

- *Prediction*: Evolutionary divergence in stomatal traits on one surface will be isometric with divergence in stomatal traits on the other surface

We tested these predictions in a phylogenetic comparative framework by compiling stomatal anatomy data from the literature for a broad range of seed plants. 	

# Methods

Unless otherwise mentioned, we performed all data wrangling and statistical analyses in *R* version `r as.character(getRversion())` [@r_core_team_r:_2021]. The source code is available to reviewers. All source code will be publically available on GitHub upon publication (https://github.com/cdmuir/stomata-independence) and archived on Zenodo.

## Data synthesis

We searched the literature for studies that measured stomatal density and stomatal size, either guard cell length or stomatal pore length, for both abaxial and adaxial leaf surfaces. In other words, we did not include studies unless they reported separate density and size values for each surface. We did not record leaf angle because it is typically not reported, but we presume that for the vast majority of taxa that the abaxial is the lower surface and the adaxial is the upper surface. This is reversed in resupinate leaves, but to the best of our knowledge, our synthesis did not include resupinate leaves. We refer to guard cell length as stomatal length and converted stomatal pore length to stomatal length assuming guard cell length is twice pore length [@sack_developmental_2016]. Table \@ref(tab:traits) lists focal traits and symbols.

```{r, traits}

tribble(
  ~Symbol, ~`Variable string`, ~Units,
  "$D_\\mathrm{ab}$", "$\\mathtt{abaxial\\_stomatal\\_density\\_mm2}$", "$\\text{pores mm}^{-2}$",
  "$D_\\mathrm{ad}$", "$\\mathtt{adaxial\\_stomatal\\_density\\_mm2}$", "$\\text{pores mm}^{-2}$",
  "$L_\\mathrm{ab}$", "$\\mathtt{abaxial\\_stomatal\\_length\\_um}$", "$\\mu$m",
  "$L_\\mathrm{ad}$", "$\\mathtt{adaxial\\_stomatal\\_length\\_um}$", "$\\mu$m"
) |>
  knitr::kable(format = "markdown", caption = "Stomatal anatomical traits with mathemtical symbol, variable string used in source code, and scientific units.", align = "ccc")

```

Data on stomatal anatomy are spread over a disparate literature and we have not attempted an exhaustive synthesis of amphistomatous leaf stomatal anatomy. We began our search by reviewing papers that cited key studies of amphistomy [@parkhurst_adaptive_1978; @mott_adaptive_1982; @muir_making_2015]. We supplemented these by searching Clarivate *Web of Science* for "guard cell length" because most studies that report guard cell length also report stomatal density, whereas the reverse is not true. We identified additional studies by reviewing the literature cited of papers we found and through haphazard discovery. The final data set contained `r nrow(stomata_anatomy1)` observations of stomatal density and length from `r length(unique(stomata_anatomy1$source_id))` taxa and `r length(unique(stomata_anatomy1$source))` primary studies (Table \@ref(tab:sources)). However, many of these data were excluded if taxonomic name and phylogenetic placement could not be resolved (see below). Finally, we included a previously unpublished data set of 14 amphistomatous wild tomato species (*Solanum* sect. *Lycopersicum* and sect. *Lycopersicoides*) grown in pots under outdoor summer Mediterranean conditions previously described [@muir_unpublished_2021]. We took ab- and adaxial epidermal imprints using clear nail polish of the mid-portion of the lamina away from major veins on the terminal leaflet of the youngest, fully expanded leaf from 1-5 replicates per taxon. With a brightfield light microscope, we counted stomata in three fields of view and divided by area to obtain density and measured the average guard cell length to estimate stomatal size. The data set is publicly available as an *R* package **ropenstomata** (https://github.com/cdmuir/ropenstomata). For review, we have included the current version of **ropenstomata** and the version used for publication will be deposited on Dryad and Zenodo upon publication.

## Phylogeny

We resolved taxonomic names using the R package **taxize** version `r packageVersion("taxize")` [@chamberlain_taxize_2013]. We queried taxonomic names supplied by the original study authors on `r str_extract(file.info("processed-data/taxize_output.rds")$ctime, "^[0-9]{4}-[0-9]{2}-[0-9]{2}")` from the following sources: `r taxize_source_ref`. We retained the maximum scoring matched name with taxize score $\ge 0.75$ (a score of 1 is a perfect match). In 5 ambiguous cases we manually curated names. Taxonomic name resolution reduced the data set from `r n_tax1` to `r n_tax2` taxa. Most taxa are different species, but some recognized subspecies and varieties are also included. All algorithms and choices are documented in the associated source code.

We used the R packages **taxonlookup** version `r packageVersion("taxonlookup")` [@pennell_simple_2016] and **V.phylomaker** version `r packageVersion("V.phylomaker")` [@jin_vphylomaker_2019] to maximize overlap between our data set and the GBOTB.extended mega-tree of seed plants [@smith_constructing_2018 ; @zanne_three_2014]. We further resolved large ($\ge4$ taxa) polytomies in `r nrow(polytomies) - 1` clades with sufficient sequence data using **PyPHLAWD** version 1.0 [@smith_pyphlawd_2019] in `r system("python3 --version", intern = TRUE)` (Python Software Foundation, https://www.python.org/). We used sequence data from the most recent GenBank Plant and Fungal sequences database division [@ouellette_database_1997]. We inferred subtree phylogenies using `r str_extract(str_c(system("raxmlHPC-PTHREADS -version", intern = TRUE), collapse = ""), "RAxML version [0-9]*.[0-9]*.[0-9]*")` [@stamatakis_raxml_2014] and conducated molecular dating using the `chronos()` function in the R package **ape** version `r packageVersion("ape")` [@paradis_ape_2019] to obtain ultrametric trees. We grafted resolved, ultrametric subtrees onto the mega-tree at the polytomy nodes and rescaled to keep the mega-tree ultrametric. In some cases, resolving polytomies was not possible because there was little or no overlap between taxa in the data set and taxa with sequence data available for **PyPHLAWD**. In these cases, we randomly selected two taxa as a phylogenetially independent pair and dropped the rest. Remaining polytomies of three taxa were resolved randomly using the `multi2di()` function in **ape**. The final data set for which we had both trait and phylogenetic information contained `r n_tax3` taxa. Seven taxa are gymnosperms; the vast majority are angiosperms. 

## Phylogenetically independent contrasts

We extracted `r n_pair` phylogenetically independent taxon pairs. A fully resolved, bifurcating four-taxon phylogeny can have two basic topologies: $\tt{((A,B),(C,D))}$ or $\tt{((A,B),C),D))}$. Taxon pairs include all comparisons of $\tt{A}$ with $\tt{B}$ and $\tt{C}$ with $\tt{D}$ in each four-taxon clade. We extracted pairs using the `extract_sisters()` function in R package **diverge** version `r packageVersion("diverge")` [@anderson_diverge_2021] and custom scripts (see source code). Taxon pairs are the most closely related pairs in our data set, but not usually the most closely related species in nature. For each pair we calculated phylogenetically independent contrasts [@felsenstein_phylogenies_1985] as the difference in the log$_{10}$-transformed trait value. Contrasts are denoted as $\Delta \text{log(trait)}$. We log-transformed traits for normality because like many morphological and anatomical traits they are strongly right-skewed. Log-transformation also helps compare density and length, which are measured on different scales, because log-transformed values quantify proportional rather than absolute divergence. We only used pairs of terminal taxa rather than the entire tree for two reasons. First, even in whole-tree methods, approximately half of the data comes from divergence at the tips, so we do not lose much statistical power. Second, using a taxon-pair method obviates making strong assumptions about multivariate trait evolution process homogeneity throughout seed plant evolution. Homogeneity is biologically unlikely and it is difficult to fit more complex models reliably with only data from extant taxa.

## Parameter estimation

Both hypotheses make predictions about trait covariance matrices or parameters derived from them (see next subsections). We estimated the $4 \times 4$ covariance matrix of phylogenetically independent contrasts between log-transformed values of $\Delta \text{log}(D_\mathrm{ab})$, $\Delta \text{log}D_\mathrm{ad})$, $\Delta \text{log}(L_\mathrm{ab})$, and $\Delta \text{log}(L_\mathrm{ad})$ using a distributional multiresponse robust Bayesian approach. We denote variances as $\text{Var}[\Delta \text{log(trait)}]$ and covariances as $\text{Cov}[\Delta \text{log(trait}_1),\Delta \text{log(trait}_2)]$. The statistical model also accounts for differences in mean trait values between surfaces. 

We used a multivariate $t$-distribution rather than a Normal distribution because estimates using the former are more robust to exceptional trait values. Exceptional trait values are common in biology and can distort estimates of central tendency and variance. The $t$-distribution is more robust because it has fatter tails (more kurtosis) than the Normal distribution. The $\nu$ parameter of the $t$-distribution describes how fat the tails are. As $\nu \to \infty$, the $t$-distribution converges to the Normal distribution. We estimated $\nu$ from the data, meaning that if exceptional values are absent from the data, the model will be nearly equivalent to standard Normal regression.

We also estimated whether the variance in trait divergence increases with time. Under many trait evolution models (e.g. Brownian motion), interspecific variance increases through time. To account for this, we included time since taxon-pair divergence as an explanatory variable affecting the trait covariance matrix, but not the trait mean. 

We fit the model in Stan `r str_extract(as.character(fit$version$cmdstan), "^2.[1-2][0-9]")` [@stan_development_team_stan_2021] using the R packages **brms** version `r as.character(fit$version$brms)` [@burkner_brms_2017; @burkner_advanced_2018] with a **cmdstanr** version `r as.character(fit$version$cmdstanr)` backend [@gabry_cmdstanr_2021]. It ran on `r length(fit$fit@stan_args)` parallel chains for `r fit$fit@stan_args[[1]][["warmup"]]` warm-up iterations and `r fit$fit@stan_args[[1]][["iter"]] - fit$fit@stan_args[[1]][["warmup"]]` sampling iterations. All parameters converged ($\hat{R} \approx 1$) and the effective sample size from the posterior exceeded 1000 [@vehtari_rank-normalization_2021]. We used the posterior median for point estimates and calculated uncertainty with the 95\% highest posterior density (HPD) interval from the posterior distribution.

## Is size-density scaling the same on both leaf surfaces?

We tested the first hypothesis by estimating the covariance between divergence in stomatal length and stomatal density on each leaf surface. If size and density are developmentally integrated, we predict the covariance matrices will not be significantly different. Specifically, the 95\% HPD intervals of the difference in covariance parameters should not include 0 if:

\begin{gather}\label{eq:prediction1}
\text{Var}[\Delta \text{log}(D_\text{ab})] \ne \text{Var}[\Delta \text{log}(D_\text{ad})] \\
\text{Var}[\Delta \text{log}(L_\text{ab})] \ne \text{Var}[\Delta \text{log}(L_\text{ad})] \\
\text{Cov}[\Delta \text{log}(L_\text{ab}), \Delta \text{log}(D_\text{ab})] \ne \text{Cov}[\Delta \text{log}(L_\text{ad}), \Delta \text{log}(D_\text{ad})]
\end{gather}

## Do abaxial and adaxial stomatal traits evolve isometrically?

If stomatal traits on each surface are developmentally integrated then divergence in the trait on one surface should result in a 1:1 (isometric) change in the trait on the other surface. Furthermore, there should be relatively little variation away from a 1:1 relationship. Conversely, if traits can evolve independently then the change in the trait on one surface should be uncorrelated with changes on the other. We tested for isometry by estimating the standardized major axis (SMA) slope of change in the abaxial trait against change in the adaxial trait for both stomatal length and stomatal density. If change on each surface is isometric, then the HPD intervals for the slope should include 1. We used the coefficient of determination, $r^2$, to quantify the strength of integration, where a value of 1 is complete integration and a value of 0 is complete disintegration. 

# Results

## Adaxial stomatal density is more variable, but size-density covariance is similar on both surfaces

Stomatal length negatively covaries with stomatal density similarly on both surfaces, but on the adaxial surface there are many more taxa that have low stomatal density and small size compared to the abaxial surface (Fig. \ref{fig:h1_raw}). In principle, this pattern could arise either because size-density covariance differs or the variance in adaxial stomatal density increases faster than that for abaxial stomatal density. Also note that the interspecific variance increases with time since divergence for all traits (Table \ref{table:model_output}), but the covariance matrix structure did not change qualitatively over time (results not shown). For consistency, we therefore report estimates conditional on time since divergence set to 0. Across pairs, we estimate that the covariance between size and density is similar. The median estimate is $\text{Cov}[\Delta \text{log}(L_\text{ad}), \Delta \text{log}(D_\text{ad})] - \text{Cov}[\Delta \text{log}(L_\text{ab}), \Delta \text{log}(D_\text{ab})] = `r dplyr::filter(h1_parameters, name == "diff_cov")[["estimate"]]`$, but 0 is within the range of uncertainty (`r dplyr::filter(h1_parameters, name == "diff_cov")[["interval"]]`). However the variance in adaxial stomatal density is significantly greater than the abaxial stomatal density [Fig. \ref{fig:h1}]). We estimate $\text{Var}[\Delta \text{log}(D_\text{ad})]$ is $`r dplyr::filter(h1_parameters, name == "diff_sigma2_d")[["estimate"]]`$ (`r dplyr::filter(h1_parameters, name == "diff_sigma2_d")[["interval"]]`) greater than $\text{Var}[\Delta \text{log}(D_\text{ab})]$. The variance in stomatal length was similar for both surfaces, with an estimate of $`r dplyr::filter(h1_parameters, name == "diff_sigma2_l")[["estimate"]]`$ (`r dplyr::filter(h1_parameters, name == "diff_sigma2_l")[["interval"]]`).


\begin{figure}[ht]
\includegraphics[width=\textwidth]{../figures/h1-raw.pdf}
\caption{Inverse size-density scaling in a synthesis of amphistomatous leaf traits across `r n_tax3` taxa. The panels show the relationship between stomatal length ($x$-axis) and stomatal density ($y$-axis) on a log-log scale for values measured on the abaxial leaf surface (left) and the adaxial leaf surface (right). To avoid overplotting of points, we used a hexbin plot in which the shade of the hexagonal bin indicates the number of points in that bin. Whiter shades indicate more points (see scale to the right).}
\label{fig:h1_raw}
\end{figure}

\begin{figure}[ht]
\includegraphics[width=\textwidth]{../figures/h1.pdf}
\caption{Evolutionary divergence in adaxial stomatal density is more variable, but covariance between density and length is similar on both surfaces. (a) Data from `r n_pair` phylogenetically independent contrasts of change in log(stomatal length) ($x$-axis) and log(stomatal density) ($y$-axis) for abaxial (left panel) and adaxial (right panel) leaf surfaces. Each contrast is shown by black points and every contrast appears on both panels. Grey ellipses are the model-estimated 95\% covariance ellipses. The negative covariance is similar for both surfaces but the breadth in the $y$-direction is larger for adaxial traits, indicating greater evolutionary divergence in log(stomatal density). (b) Parameter estimates (points), 66\% (thick lines), and 95\% HPD intervals for estimates of trait (co)variance. Grey points and lines represent ab- and adaxial values; black points and lines represent the estimated difference in (co)variance between surfaces. Only the variance for stomatal density (middle panel) is significantly greater for the adaxial surface (95\% HPD interval does not overlap the dashed line at 0). Reported paremeter estimates are conditioned on zero time since divergence between taxa (see \protect\hyperlink{results}{Results}).}
\label{fig:h1}
\end{figure}

## Stomatal density on each surface is less integrated than stomatal length

The relationship between stomatal density on each leaf surface is visually more variable than that for stomatal length (Fig. \ref{fig:h2_raw}). This pattern occurs because the slope and strength of integration for stomatal density on each surface is much weaker than that for stomatal length. The SMA slope between $\Delta \text{log}(D_\text{ad})$ and $\Delta \text{log}(D_\text{ab})$ is less than 1 (estimated slope $= `r dplyr::filter(h2_parameters, name == "slope_density")[["estimate"]]`$, `r dplyr::filter(h2_parameters, name == "slope_density")[["interval"]]`) and the strength of association is weakly positive (estimated $r^2 = `r dplyr::filter(h2_parameters, name == "r2_density")[["estimate"]]`$, `r dplyr::filter(h2_parameters, name == "r2_density")[["interval"]]`; Fig. \ref{fig:h2}). In contrast, the relationship between $\Delta \text{log}(L_\text{ad})$ and $\Delta \text{log}(L_\text{ab})$ is isometric (estimated slope $= `r dplyr::filter(h2_parameters, name == "slope_length")[["estimate"]]`$, `r dplyr::filter(h2_parameters, name == "slope_length")[["interval"]]`) and strongly positive (estimated $r^2 = `r dplyr::filter(h2_parameters, name == "r2_length")[["estimate"]]`$, `r dplyr::filter(h2_parameters, name == "r2_length")[["interval"]]`; Fig. \ref{fig:h2}).

\begin{figure}[ht]
\includegraphics[width=\textwidth]{../figures/h2-raw.pdf}
\caption{Relationship stomatal density and length on each leaf surface in a synthesis of amphistomatous leaf traits across `r n_tax3` taxa. The panels show the relationship between the abaxial trait value ($x$-axis) and the adaxial trait value ($y$-axis) on a log-log scale for stomatal density (left) and stomatal length (right). To avoid overplotting of points, we used a hexbin plot in which the shade of the hexagonal bin indicates the number of points in that bin. Whiter shades indicate more points (see scale to the right). The dashed line in across the middle is the 1:1 line for reference.}
\label{fig:h2_raw}
\end{figure}

\begin{figure}[ht]
\includegraphics[width=\textwidth]{../figures/h2.pdf}
\caption{Developmental integration in stomatal length is much stronger than stomatal density between the surfaces of amphistomatous leaves (a) Data from `r n_pair` phylogenetically independent contrasts of change in the abaxial trait value ($x$-axis) against change in the adaxial trait value ($y$-axis) for log(stomatal density) ($x$-axis) and log(stomatal density) (left panel) and log(stomatal length) (right panel). Each contrast is shown by black points and every contrast appears on both panels. Dashed grey lines are 1:1 lines for reference. Solid grey lines and ribbon the fitted SMA slope and 95\% HPD interval. (b) The SMA slope (left panel) is significantly less than 1 (isometry, top dashed line) for density but very close to isometric for length. The coefficient of determination ($r^2$, right panel) is also much greater for length than density. The points are parameter estimates with 66\% (thick lines) and 95\% HPD intervals. Reported paremeter estimates are conditioned on zero time since divergence between taxa (see \protect\hyperlink{results}{Results}).}
\label{fig:h2}
\end{figure}

# Discussion

Developmental integration leads to trait covariation and may hinder adaptation by preventing traits from evolving independently toward a multivariate phenotypic optimum. Two major features of stomatal anatomical variation at the macroevolutionary scale, inverse size-density scaling and bimodal stomatal ratio, may be shaped by developmental integration between pavement and guard cell size on both leaf surfaces. In this study, we took advantage of the fact that amphistomatous leaves produce stomata on both abaxial (usually lower) and adaxial (usually upper) surfaces to test predictions of developmental integration hypotheses using a global phylogenetic comparison of seed plants, albeit mostly angiosperms. Contrary to developmental integration hypotheses, adaxial stomatal density ($D_\text{ad}$) evolves somewhat independently of adaxial stomatal length ($L_\text{ad}$) and abaxial stomatal density ($D_\text{ab}$). Hence, inverse size-density scaling and bimodal stomatal ratio cannot be attributed entirely to developmental integration. Quasi-independent evolution of traits should enable lineages to reach their fitness optimum, implying that selection is most likely a major constraint on the variation in stomatal anatomy in seed plants. Future research should test whether developmental integration is more or less important in certain lineages and identify the mechanistic basis of developmental disintegration where it occurs on the plant tree of life.

## Does developmental integration lead to inverse size-density scaling?

If stomatal size and density are both determined by meristematic cell volume [@brodribb_unified_2013], then we predicted inverse size-density scaling would evolve with the same (co)variance for both ab- and adaxial leaf surfaces. Contrary to this prediction, there are many combinations of stomatal density and length found on adaxial leaf surfaces that are absent from abaxial leaf surfaces (Fig. \ref{fig:h1_raw}). In principle, the different relationship between traits on each surface could be caused by different evolutionary variance in stomatal density ($\text{Var}[\Delta \text{log}(D_\text{ab})] \ne \text{Var}[\Delta \text{log}(D_\text{ad})]$) and/or covariance ($\text{Cov}[\Delta \text{log}(L_\text{ab}), \Delta \text{log}(D_\text{ab})] \ne \text{Cov}[\Delta \text{log}(L_\text{ad}), \Delta \text{log}(D_\text{ad})]$) on each surface. However, the covariance relationship between density and length is similar on each surface, whereas the evolutionary variance in adaxial stomatal density is significantly higher than that for abaxial density ($\text{Var}[\Delta \text{log}(D_\text{ab})] < \text{Var}[\Delta \text{log}(D_\text{ad})]$; Fig. \ref{fig:h1}). Given that the average stomatal length is usually about the same on each surface (see below), these results imply that plants can often evolve stomatal densities on each surface without a concomitant change in size. 

The disintegration of stomatal size and density on adaxial leaf surfaces implies that the inverse size-density scaling on abaxial surfaces [@weiss_untersuchungen_1865; @franks_maximum_2009; @de_boer_optimal_2016; @sack_developmental_2016; @liu_scaling_2021] is not a developmental *fait accompli*. The lability of $D_\text{ad}$ may explain why there is so much putatively adaptive variation in the trait along light gradients [@muir_light_2018] and in coordination with other anatomical traits [@pathare_increased_2020]. It also suggests that the relationship between genome size and stomatal anatomy at macroevolutionary scales [@roddy_scaling_2020] may not be causal. Genome size sets a minimum on meristematic cell volume [@simova_geometrical_2012], but the decoupling of size and density on the adaxial surface suggests that meristematic cell volume is probably not a strong constraint on the final size of pavement and guard cells.

## Does developmental integration lead to bimodal stomatal ratio?

We predicted that if abaxial and adaxial stomata are developmentally integrated then we should observe a strong, isometric relationship between trait divergence on each surface. Consistent with this prediction, divergence in stomatal length on each surface is isometric (SMA slope $=$ `r dplyr::filter(h2_parameters, name == "slope_length")[["estimate"]]`) and strongly associated ($r^2 =$ `r dplyr::filter(h2_parameters, name == "r2_length")[["estimate"]]`; Fig. \ref{fig:h2}). In contrast, divergence in stomatal density on each surface was not isometric (SMA slope $=$ `r dplyr::filter(h2_parameters, name == "slope_density")[["estimate"]]`) and much less integrated ($r^2 =$ `r dplyr::filter(h2_parameters, name == "r2_density")[["estimate"]]`; Fig. \ref{fig:h2}). Since average stomatal density on each surface can evolve quasi-independently, a wide variety of stomatal ratios are developmentally possible. Leaves are not "forced" to deploy identical stomatal development programs on each surface. Hence, the bimodal stomatal ratio pattern [@muir_making_2015] is unlikely to be the result of developmental integration alone.

## Limitations and future research

The ability of adaxial stomatal density to evolve independently of stomatal size and abaxial stomatal density is not consistent with the hypothesis that developmental integration is the primary cause leading to inverse size-density scaling or bimodal stomatal ratio. However, there are two major limitations of this study that should be addressed in future work. First, while $D_\text{ab}$ can diverge independently of other stomatal traits globally, we cannot rule out that developmental integration is important in some lineages. For example, Berg's rule observes that vegetative and floral traits are often developmentally integrated, but integration can be broken when selection favors flowers for specialized pollination [@berg_general_1959; @berg_ecological_1960; @conner_raissa_2014]. Analogously, developmental integration between stomatal anatomical traits could evolve in some lineages, due to selection or other evolutionary forces, but become less integrated in other lineages. For example, $D_\text{ab}$ and $D_\text{ad}$ are positively genetically correlated in *Oryza* [@ishimaru_identification_2001; @rae_elucidating_2006], suggesting developmental integration may contribute to low variation in stomatal ratio between species of this genus [@giuliani_coordination_2013]. A second major limitation is that covariation in traits like stomatal length, which appear to be developmentally integrated on each surface, could be caused by other processes. For example, since stomatal size affects the speed and mechanics of stomatal closure [@drake_smaller_2013; @harrison_influence_2020], there may be strong selection for similar stomatal size throughout the leaf to harmonize rates of stomatal closure. Coordination between epidermal and mesophyll development may also constrain how independently stomatal traits on each surface can evolve [@dow_disruption_2017; @lundgren_mesophyll_2019].

Future research should identify the mechanistic basis of developmental disintegration between $D_\text{ab}$ and $D_\text{ad}$. Multiple reviews of stomatal development conclude that stomatal traits are independently controlled on each surface [@lake_longdistance_2002; @bergmann_stomatal_2007], but we do not know much about linkage between ab-adaxial polarity and stomatal development [@kidner_signaling_2010; @pillitteri_mechanisms_2012]. Systems that have natural variation in stomatal ratio should allow us to study how developmental disintegration evolves. Quantitative genetic studies in *Brassica oleracea* L., *Oryza sativa* L., *Populus trichocarpa* Torr. \& A. Gray ex Hook., *Populus* interspecific crosses, and *Solanum* interspecific crosses, typically find partial independence of $D_\text{ab}$ and $D_\text{ad}$; some loci affect both traits, but some loci only affect density on one surface and/or genetic correlations are weak [@ishimaru_identification_2001; @ferris_leaf_2002; @hall_relationships_2005; @rae_elucidating_2006; @laza_quantitative_2010; @chitwood_quantitative_2013; @mckown_association_2014; @muir_quantitative_2014; @porth_evolutionary_2015; @fetter_growthdefense_2021]. For example, *Populus trichocarpa* populations have putatively adaptive genetic variation in $D_\text{ad}$. Populations are more amphistomatous at Northern latitudes with shorter growing seasons that may select for faster carbon assimilation [@mckown_association_2014; @kaluthota_higher_2015; @porth_evolutionary_2015]. Genetic variation in key stomatal development transcription factors is associated with latitudinal variation in $D_\text{ad}$, which should help reveal mechanistic basis of developmental disintegration between surfaces [@mckown_role_2019].

<!-- CONCLUSION PARAGRAPH? -->

\clearpage

# Supporting Information

\renewcommand\thefigure{S\arabic{figure}}    
\renewcommand\thetable{S\arabic{table}}    
\setcounter{figure}{0}    
\setcounter{table}{0}    

```{r, sources}
library(ropenstomata)

stomata_anatomy_metadata |>
  dplyr::mutate(Source = stringr::str_c("@", source), Taxa = taxa) |>
  dplyr::select(Source, Taxa) |>
  knitr::kable(format = "markdown", caption = "Primary sources of stomatal anatomical data and the taxa covered by each source.")

```

\clearpage

# References
